lib/ansiblelint/__init__.py

```diff
@@ -19,167 +19,9 @@
-        self.verbosity = verbosity
-        if checked_files is None:
-            checked_files = set()
-        self.checked_files = checked_files
-
-    def _update_exclude_paths(self, exclude_paths):
-        if exclude_paths:
-            # These will be (potentially) relative paths
-            paths = ansiblelint.utils.expand_paths_vars(exclude_paths)
-            # Since ansiblelint.utils.find_children returns absolute paths,
-            # and the list of files we create in `Runner.run` can contain both
-            # relative and absolute paths, we need to cover both bases.
-            self.exclude_paths = paths + [os.path.abspath(p) for p in paths]
-        else:
-            self.exclude_paths = []
-
-    def is_excluded(self, file_path):
-        # Any will short-circuit as soon as something returns True, but will
-        # be poor performance for the case where the path under question is
-        # not excluded.
-        return any(file_path.startswith(path) for path in self.exclude_paths)
-
-    def run(self) -> List:
-        files = list()
-        for playbook in self.playbooks:
-            if self.is_excluded(playbook[0]) or playbook[1] == 'role':
-                continue
-            files.append({'path': ansiblelint.utils.normpath(playbook[0]),
-                          'type': playbook[1],
-                          # add an absolute path here, so rules are able to validate if
-                          # referenced files exist
-                          'absolute_directory': os.path.dirname(playbook[0])})
-        visited: Set = set()
-        matches = list()
-
-        while (visited != self.playbooks):
-            for arg in self.playbooks - visited:
-                try:
-                    for child in ansiblelint.utils.find_children(arg, self.playbook_dir):
-                        if self.is_excluded(child['path']):
-                            continue
-                        self.playbooks.add((child['path'], child['type']))
-                        files.append(child)
-                except MatchError as e:
-                    e.rule = LoadingFailureRule
-                    matches.append(e.get_match())
-                visited.add(arg)
-
-        # remove duplicates from files list
-        files = [value for n, value in enumerate(files) if value not in files[:n]]
-
-        # remove files that have already been checked
-        files = [x for x in files if x['path'] not in self.checked_files]
-        for file in files:
-            _logger.debug(
-                "Examining %s of type %s",
-                ansiblelint.utils.normpath(file['path']),
-                file['type'])
-            matches.extend(self.rules.run(file, tags=set(self.tags),
-                           skip_list=self.skip_list))
-        # update list of checked files
-        self.checked_files.update([x['path'] for x in files])
-
-        return matches
+default_rulesdir = os.path.join(os.path.dirname(__file__), 'rules')
```

Comment:
It seems weird that this var is declared right here.

Reply:
@webknjaz I agree with you but that is also outside the scope. Likely is kept here in order to allow some consumers to override it. We can think about changing this in different PR.